go/feast/featurestore.go

```diff
@@ -0,0 +1,907 @@
+	fs.dropUnneededColumns(onlineFeatureResponse, requestedResultRowNames)
+	return onlineFeatureResponse, nil
+}
+
+func (fs *FeatureStore) DestructOnlineStore() {
+	fs.onlineStore.Destruct()
+}
+
+func (fs *FeatureStore) parseKind(kind interface{}) (interface{}, error) {
+	if featureList, ok := kind.(*serving.GetOnlineFeaturesRequest_Features); ok {
+		return featureList.Features.GetVal(), nil
+	}
+	if featureServiceRequest, ok := kind.(*serving.GetOnlineFeaturesRequest_FeatureService); ok {
+		featureService, err := fs.registry.getFeatureService(fs.config.Project, featureServiceRequest.FeatureService)
+		if err != nil {
+			return nil, err
+		}
+		return featureService, nil
+	}
+	return nil, errors.New("Cannot parse kind from GetOnlineFeaturesRequest")
+}
+
+/*
+	This function returns all feature references from GetOnlineFeaturesRequest.
+	If a list of feature references is passed, return it.
+	Otherwise, FeatureService was passed, parse this feature service
+	to get a list of FeatureViewProjection and return feature references
+	from this list
+*/
+
+func (fs *FeatureStore) getFeatures(parsedKind interface{}, allowCache bool) ([]string, error) {
+
+	if features, ok := parsedKind.([]string); !ok {
+		var featureService *FeatureService
+		if featureService, ok = parsedKind.(*FeatureService); !ok {
+			return nil, errors.New("Cannot parse FeatureService from request")
+		}
+
+		var featureViewName string
+		features = make([]string, 0)
+		for _, featureProjection := range featureService.projections {
+			featureViewName = featureProjection.nameToUse()
+			for _, feature := range featureProjection.features {
+				features = append(features, fmt.Sprintf("%s:%s", featureViewName, feature.name))
+			}
+		}
+		return features, nil
+	} else {
+		return features, nil
+	}
+}
+
+/*
+	If features passed into GetOnlineFeaturesRequest as a list of feature references,
+		return all FeatureView, OnDemandFeatureView, RequestFeatureView from the registry
+	Otherwise, a FeatureService was passed, return a list of copies of FeatureViewProjection
+		copied from FeatureView, OnDemandFeatureView, RequestFeatureView existed in the registry
+
+	TODO (Ly): Since the implementation of registry has changed, a better approach here is just
+		retrieving featureViews asked in the passed in list of feature references instead of
+		retrieving all feature views. Similar argument to FeatureService applies.
+
+*/
+func (fs *FeatureStore) getFeatureViewsToUse(parsedKind interface{}, allowCache, hideDummyEntity bool) ([]*FeatureView, []*RequestFeatureView, []*OnDemandFeatureView, error) {
+
+	fvs := make(map[string]*FeatureView)
+	requestFvs := make(map[string]*RequestFeatureView)
+	odFvs := make(map[string]*OnDemandFeatureView)
+
+	featureViews, err := fs.listFeatureViews(allowCache, hideDummyEntity)
+	if err != nil {
+		// TODO (Ly): Review: return an empty list instead of
+		// error if no FeatureView is found in this project
+		featureViews = make([]*FeatureView, 0)
```

Comment:
So what response would the user see in this case? Would they understand what is happening?

Reply:
I'm adding a RegistryStore like python sdk so it returns error (exception in calling GetOnlineFeatures from go server) if call to refresh/ initialize registry is not successful instead